superset/connectors/base/models.py

```diff
@@ -213,6 +235,77 @@ def data(self) -> Dict[str, Any]:
             "select_star": self.select_star,
         }
 
+    def data_for_slices(self, slices: List[Slice]) -> Dict[str, Any]:
+        """
+        The representation of the datasource containing only the required data
+        to render the provided slices.
+
+        Used to reduce the payload when loading a dashboard.
+        """
+        data = self.data
+        metric_names = set()
+        column_names = set()
+        for slc in slices:
+            form_data = slc.form_data
+
+            # pull out all required metrics from the form_data
+            for param in METRICS_FORM_DATA_PARAMS:
+                for metric in form_data.get(param) or []:
+                    metric_names.add(utils.get_metric_name(metric))
+
+                    # extra logic to handle adhoc metrics with column references
+                    if isinstance(metric, dict) and metric.get("column"):
+                        column_names.add(metric.get("column", {}).get("column_name"))
+
+            for param in METRIC_FORM_DATA_PARAMS:
+                metric = form_data.get(param)
+                metric_names.add(utils.get_metric_name(metric))
+
+            # pull out all required columns from the form_data
+            for filter_ in form_data.get("adhoc_filters") or []:
+                if filter_["clause"] == "WHERE" and filter_.get("subject"):
+                    column_names.add(filter_.get("subject"))
+
+            for param in COLUMNS_FORM_DATA_PARAMS:
+                for column in form_data.get(param) or []:
+                    column_names.add(column)
+
+            for param in COLUMN_FORM_DATA_PARAMS:
+                if form_data.get(param):
+                    column_names.add(form_data.get(param))
+
+        filtered_metrics = [
+            metric
+            for metric in data["metrics"]
+            if metric["metric_name"] in metric_names
+        ]
+
+        filtered_columns = [
+            column
+            for column in data["columns"]
+            if column["column_name"] in column_names
+        ]
+
+        del data["description"]
+        data.update({"metrics": filtered_metrics})
+        data.update({"columns": filtered_columns})
+        verbose_map = {"__timestamp": "Time"}
+        verbose_map.update(
+            {
+                o["metric_name"]: o["verbose_name"] or o["metric_name"]
+                for o in filtered_metrics
+            }
+        )
+        verbose_map.update(
+            {
+                o["column_name"]: o["verbose_name"] or o["column_name"]
```

Comment:
Could we use `column` or `col` rather than `o`?

Reply:
yup, this was copied from above, but  agree a real name is better